pi_mqtt_gpio/server.py

```diff
@@ -505,7 +505,7 @@ def sensor_timer_thread(SENSOR_MODULES, sensor_inputs, topic_prefix):
                 sensor = SENSOR_MODULES[sens_conf["module"]]
 
                 try:
-                    value = round(sensor.get_value(sensor), sens_conf["digits"])
+                    value = round(sensor.get_value(sensor,sens_conf["name"]), sens_conf["digits"])
```

Comment:
This is the part I have an issue with. As you mention in your example config, the names 'temperature' and 'humidity' are unable to change. This is a problem if you have, for example, multiple DHT sensors, or other sensors already named 'temperature' or 'humidity'.

I think I'd prefer to add a config value for this, instead of assigning hidden meaning to the name itself.

Reply:
Perhaps something like this would work:

```yaml
sensor_modules:
  - name: dht22
    module: dht22
    sensor_type: AM2302 # can be  DHT11, DHT22 or AM2302
    pin: 4
sensor_inputs:
  - name: temperature
    module: dht22
    interval: 10 #interval in seconds, that a value is read from the sensor and a update is published
    digits: 4 # number of digits to be round
    options:
      type: temperature
  - name: humidity
    module: dht22
    interval: 10 #interval in seconds, that a value is read from the sensor and a update is published
    digits: 4 # number of digits to be round
    options:
      type: humidity
```

and then pass all of the `options` dict to the `get_value()` of the module.